pkg/controllers/configobservation/configobservercontroller/apply_configuration_test.go

```diff
@@ -0,0 +1,140 @@
+	"github.com/openshift/library-go/pkg/manifestclient"
+	libgoetcd "github.com/openshift/library-go/pkg/operator/configobserver/etcd"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/genericoperatorclient"
+	"github.com/openshift/library-go/pkg/operator/resourcesynccontroller"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+func TestApplyConfiguration(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	mutationTrackingHTTPClient := manifestclient.NewHTTPClient("/home/deads/Downloads/must-gather-01")
+
+	kubeClient, err := kubernetes.NewForConfigAndClient(&rest.Config{}, mutationTrackingHTTPClient.GetHTTPClient())
+	if err != nil {
+		t.Fatal(err)
+	}
+	configClient, err := configclient.NewForConfigAndClient(&rest.Config{}, mutationTrackingHTTPClient.GetHTTPClient())
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	operatorClient, dynamicInformers, err := genericoperatorclient.NewOperatorClientWithClient(
+		mutationTrackingHTTPClient.GetHTTPClient(),
+		operatorv1.GroupVersion.WithResource("authentications"),
+		operatorv1.GroupVersion.WithKind("Authentication"),
+		extractOperatorSpec,
+		extractOperatorStatus,
+	)
+
+	eventRecorder := events.NewLoggingEventRecorder("cluster-authentication-operator")
+
+	configInformers := configinformer.NewSharedInformerFactoryWithOptions(configClient, 0)
+	kubeInformersForNamespaces := v1helpers.NewKubeInformersForNamespaces(
+		kubeClient,
+		"default",
+		"openshift-authentication",
+		"openshift-config",
+		"openshift-config-managed",
+		"openshift-oauth-apiserver",
+		"openshift-authentication-operator",
+		"", // an informer for non-namespaced resources
+		"kube-system",
+		libgoetcd.EtcdEndpointNamespace,
+	)
+	resourceSyncer := resourcesynccontroller.NewResourceSyncController(
+		"oauth-server",
+		operatorClient,
+		kubeInformersForNamespaces,
+		v1helpers.CachedSecretGetter(kubeClient.CoreV1(), kubeInformersForNamespaces),
+		v1helpers.CachedConfigMapGetter(kubeClient.CoreV1(), kubeInformersForNamespaces),
+		eventRecorder,
+	)
+
+	clusterVersion, err := configClient.ConfigV1().ClusterVersions().Get(ctx, "version", metav1.GetOptions{})
+	if err != nil {
+		t.Fatal(err)
+	}
+	enabledClusterCapabilities := sets.NewString()
+	for _, v := range clusterVersion.Status.Capabilities.EnabledCapabilities {
+		enabledClusterCapabilities.Insert(string(v))
+	}
+
+	configObserver := NewConfigObserver(
+		operatorClient,
+		kubeInformersForNamespaces,
+		configInformers,
+		resourceSyncer,
+		enabledClusterCapabilities,
+		eventRecorder,
+	)
+	configInformers.Start(ctx.Done())
+	kubeInformersForNamespaces.Start(ctx.Done())
+	dynamicInformers.Start(ctx.Done())
+
+	// wait for cache sync (nearly instant from disk)
```

Comment:
why not to use `HasSynced` ?

Reply:
it didn't exist on all our informer types. I'm forcing that into library-go now.